packages/kit/src/runtime/server/endpoint.js

```diff
@@ -77,13 +77,16 @@ export async function render_endpoint(event, mod, state) {
  * @param {import('types').RequestEvent} event
  */
 export function is_endpoint_request(event) {
-	const { method } = event.request;
+	const { method, headers } = event.request;
 
 	if (method === 'PUT' || method === 'PATCH' || method === 'DELETE') {
 		// These methods exist exclusively for endpoints
 		return true;
 	}
 
+	// use:enhance uses a custom header to disambiguate
+	if (headers.get('x-sveltekit-action') === 'true') return false;
```

Comment:
i'm not sure what it would mean for this to apply to `GET` requests

```suggestion
	if (method === 'POST' && headers.get('x-sveltekit-action') === 'true') return false;
```